### ISSUE TITLE
set `locked` in `config.toml` or `pyproject.toml`

### ISSUE THREAD

**Issue by adrinjalali** [prefix-dev/pixi#2628]
I'd like to be able to set `locked=true` on the repo so that the lock file only gets updated when intended. Right now one can only pass that as an argument to the commands like `pixi run -e ci-sklearn15 --locked tests` (context: https://github.com/skops-dev/skops/pull/451).

Also, right now when passing `--locked` and the env config is changed, it says 

```
$ pixi run -e ci-sklearn15 --locked tests
  × lock-file not up-to-date with the project
```

It'd be nice to either have an option which only updates the lock file when the environment info is changed, or to ask the user for confirmation.

**Comment by ruben-arts** [comment]
Hey @adrinjalali, 

I assume what you're looking for is `--frozen` that will not try to update the lockfile and will continue as previously installed.

**Comment by adrinjalali** [comment]
Yes, but it still cannot be set in a configuration file, which I'd need if I'm to share a project with new collaborators.

**Comment by adrinjalali** [comment]
I'd like to have a setting which would only touch my lock file if I explicitly tell it to, otherwise I'd like it to be such that no command has a _side effect_ of changing the lock file. I can then have a bot or manually update the lock file at desired intervals.

**Comment by adrinjalali** [comment]
As an example of why this is needed: https://github.com/sklearn-compat/sklearn-compat/pull/15

That PR should have been very few lines of diff, and the pixi version in the GH action shouldn't have needed updating. But since I use a newer pixi, the lock file was massively updated, and I had to update the version in the GH action at the same time. That makes very irrelevant changes to that PR.

**Comment by ruben-arts** [comment]
I found some more issues with the workflow you are using, we're looking into how we can make pixi more workable with this use case. There is a conversation about the lockfile going on internally, we've not found a good solution yet, but we'll share some ideas soon. 

One of the biggest problems with your workflow and pixi is that fact that we depend on the `pyproject.toml` to validate the lockfile which can change if you make a `rc` release often without really requiring a rebuild for the code.

More info about this later!

**Comment by adrinjalali** [comment]
@ruben-arts this feels quite odd and not a good experience to have all my open issues just closed as not planned w/o any information as really why. The closed issues are covering quite a few different areas and it seems you're simply closing doors to reports alltogether?

cc @wolfv

**Comment by ruben-arts** [comment]
> @ruben-arts this feels quite odd and not a good experience to have all my open issues just closed as not planned w/o any information as really why. The closed issues are covering quite a few different areas and it seems you're simply closing doors to reports alltogether?
> 
> cc @wolfv 

I'm sorry. This was an issue with our issue tracking tool Linear, it has suddenly closed all issues that it noted as stale because it was 6mo old. We're working on figuring out what was incorrectly closed.

**Comment by ruben-arts** [comment]
That said, this issue might become obsolete with the proposal in https://github.com/prefix-dev/pixi/issues/4457.

Do you agree?

**Comment by adrinjalali** [comment]
IIUC, that one avoids having a lockfile alltogether, here I'm asking for only updating the file when I explicitly want to.

**Comment by ogrisel** [comment]
@ruben-arts would you consider reopening this feature request? We recently considered using pixi to manage the lock files and environments of our CI for the scikit-learn project (https://github.com/scikit-learn/scikit-learn/pull/34212) and this missing configuration options is a blocker for adopting pixi.

**Comment by wolfv** [comment]
@ogrisel sure! we can reopen - maybe you guys can flesh out a little bit what the usage should be like here?

**Comment by adrinjalali** [comment]
We'd like to be able to set the option on `pyproject.toml`, so that when contributors work with the project, and whatever `pixi` command they run (except very explicit commands which would point to updating the lock file), the lock file should stay frozen.

We have a workflow where once a week a bot would update the lock file for us and automatically open a PR with the updated lock file; ideally the bot would update the lock file in sections, so that each bot PR tackles a section of the CI, but that's another story.

So the minimal feature here would be to be able to expose the same feature as `--frozen` via the `pyproject.toml` config.

**Comment by wolfv** [comment]
This is more or less the behavior currently, except when the user _changes the pixi.toml_ dependencies. It would be quite obvious if someone commits a change to the lockfile (or a change to the `pixi.toml` which would fail the `--locked` check in CI).

Did you notice problems with the current pixi or is this more a precautionary measure? :)

**Comment by ogrisel** [comment]
> Did you notice problems with the current pixi or is this more a precautionary measure? :)

Actually, I have not tried to use pixi recently because I did not know that the default behavior had changed and I thought we would still need that option for scikit-learn. Maybe the new behavior is indeed compatible with our workflow.
